src/modules/vtol_att_control/vtol_type.cpp

```diff
@@ -234,7 +234,7 @@ void VtolType::resetTransitionStates()
 {
 	_transition_start_timestamp = hrt_absolute_time();
 	_time_since_trans_start = 0.f;
-	_local_position_z_start_of_transition = _local_pos->z;
+	_local_position_z_start_of_transition = _local_pos->z - _local_pos->delta_z; // "raw" local position, so without resets
```

Comment:
The resets are not cumulative, so to be exact, you should apply it when it occurs, such as in https://github.com/PX4/PX4-Autopilot/blob/5efcde74127e75385aae858fb567717000b3fd5d/src/modules/flight_mode_manager/tasks/FlightTask/FlightTask.cpp#L61-L64

Reply:
Okay, it wasn't clear to me how these deltas are to be interpreted. I've added now comments in the message definitions.